src/api-service/tests/test_secrets.py

```diff
@@ -100,12 +100,40 @@ def test_roundtrip_github_issue(self) -> None:
             f"{current_path}"
             + "/../../../contrib/onefuzz-job-github-actions/github-issues.json"
         ) as json_file:
-            b = json.load(json_file)
-            b["container"] = "testing"
-            c = NotificationCreate.parse_obj(b)
-            d = c.json()
-            e = json.loads(d)
-            NotificationCreate.parse_obj(e)
+            notification_dict = json.load(json_file)
+            notification_dict["container"] = "testing"
+            notification1 = NotificationCreate.parse_obj(notification_dict)
+            self.assertIsInstance(
+                notification1.config, GithubIssueTemplate, "invalid config type"
+            )
+            notification1_config = cast(GithubIssueTemplate, notification1.config)
+            self.assertIsInstance(
+                notification1_config.auth.secret, GithubAuth, "Invalid secret type"
+            )
```

Comment:
I am concerned that `cast` will hide errors from mypy in the future.

Using `assertIsInstance` makes the unit test output better and is the better choice in most cases, but using a raw `assert` here enables `mypy` to verify the data structures for us without `cast`.
 
```suggestion
            notification1 = NotificationCreate.parse_obj(notification_dict)
            assert isinstance(notification1.config, GithubIssueTemplate)
            self.assertIsInstance(
                notification1.config.auth.secret, GithubAuth, "Invalid secret type"
            )
```